File: dsa-problems/Tries/solutions/java/TRI-005-binary-trie-min-xor-pair-limit.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int L = sc.nextInt();
        
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        
        int minXor = Integer.MAX_VALUE;
        
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int xorVal = arr[i] ^ arr[j];
                if (xorVal <= L) {
                    minXor = Math.min(minXor, xorVal);
                }
            }
        }
        
        System.out.println(minXor == Integer.MAX_VALUE ? -1 : minXor);
        sc.close();
    }
}

File: dsa-problems/Tries/solutions/javascript/TRI-005-binary-trie-min-xor-pair-limit.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

const lines = [];
rl.on('line', (line) => lines.push(line.trim()));
rl.on('close', () => {
    const [n, L] = lines[0].split(' ').map(Number);
    const arr = lines[1].split(' ').map(Number);
    
    let minXor = Infinity;
    
    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            const xorVal = arr[i] ^ arr[j];
            if (xorVal <= L) {
                minXor = Math.min(minXor, xorVal);
            }
        }
    }
    
    console.log(minXor === Infinity ? -1 : minXor);
});
